### DISCUSSION TITLE
Is there no longer qtile in the Debian and Fedora repositories ?

### DISCUSSION THREAD

**Discussion by r-gorajski** [qtile/qtile#3101] [General]
Today I wanted to install qtile - unfortunately searching for the package gives no results.

Fedora 35
Debian testing

dnf search qtile
apt-cache search qtile

They show nothing?

**Comment by elParaguayo** [comment]
I think someone has been looking at packaging this for Fedora but it's not ready yet. Not sure about Debian.

You could always manually install from pip or git.

**Comment by cr1ogen** [comment]
In Debian there was Qtile in the repositories, but they removed it a while ago

**Reply by hamensman** [reply]
Why on earth did they do that?

**Comment by xarvh** [comment]
Same in Ubuntu.

As much as I love qtile, every time I need to install it in a new system it's a pain and takes me days.

Right now I can't run it on any of my computers and for different reasons (one has the usual broken libraries, one gives a weird error https://github.com/qtile/qtile/issues/1040 )

Are all those problems because Python packaging/libraries system is not so great, or is it a problem with Qtile?

**Comment by hamensman** [comment]
This thread mentions how to install Qtile on Debian: https://forum.siduction.org/index.php?topic=8467.msg68052#msg68052
You have to install from source.

**Reply by cr1ogen** [reply]
Here too

http://docs.qtile.org/en/stable/manual/install/ubuntu.html

**Comment by xarvh** [comment]
Tried to install from sources.
Doesn't work.
I'll wait Ubuntu 22.04 then try again, I'm burned out now and I need to actually get stuff done.
Thanks for the help.

**Reply by hamensman** [reply]
What do you mean it doesn't work, specifically?

**Comment by port19x** [comment]
I have called for debian packaging about two months ago #3358 
Since then I have transitioned to a competing window manager, as I value portability of configuration and don't see the situation improving anytime soon.
I retract my offer to do any groundwork on this.
Burrying feature requests and any non-bug reports in github discussions leads nowhere.
If user count is at all a metric the devs cared about, packaging should be a lot higher of a priority.
Can confirm installing from source is a massive pita. I don't want to imagine how much more of a pita it is for people that are less technical

**Reply by m-col** [reply]
Unfortunately it seems that nobody with the know-how or who uses debian contributed to that discussion, and there isn't much we can do to facilitate debian packaging. AFAICT most packagers are users of both Qtile and the specific distro that they are packaging for - and nobody who uses debian has stepped forward to offer to package it. As this is an issue for you, I totally understand, and it is a shame. However the reality is that we are a small team working on this project and so we focus our time on fixing the (many) bugs it has. The best we can do for everything else is to be open for contributions for feature requests. This is why we have segregated feature requests to discussions - it makes our work easier, which is mostly fixing the bugs. FWIW we are happy to (and we do) get packaging related issues in the issues tracker, which is why the issue template says "Our tracker for bugs and other issues with the core project". Perhaps we should clarify what constitutes 'other issues'.

**Reply by elParaguayo** [reply]
To add to what mcol said (and what I said in our earlier discussion), we are very happy to try and help with packaging questions (why won't it build, why are tests failing etc). Similarly, if there are errors in the documentation, we'll happily fix them. However, we won't take the lead on the actual packaging because we need to focus our efforts elsewhere.

I would hope that installing from source works for most users so if things are going wrong there, we should look into that.

**Reply by xarvh** [reply]
@m-col @elParaguayo that's very understandable.

However, I think that the problem goes a bit deeper than that.
I would be happy to help with the packaging, but I don't think I can do that when I can't even run Qtile from sources.

If it was straightforward to run Qtile from sources, .dev packaging wouldn't be a problem, and I suspect that the same would be true for .rpm.

So maybe we should open a more specific issue "Running Qtile from sources if very difficult"?

And I'm not sure whether it is a problem specific to Qtile, or it's just that Python packaging and `pip` are not very good, but at least in my experience it is by far my major pain point with Qtile for the several years I've been using it.

**Reply by elParaguayo** [reply]
I'm with you on that. 

Qtile should be installable from source. We should be clear on how to do that and helping fixing issues related to that where we can.

If we get that right, packaging should hopefully be easier.

What I don't know is whether there is an issue with certain distros packages that are incompatible with our code.

**Reply by xarvh** [reply]
I'll try one more installation from sources and open a detailed issue.

**Comment by xarvh** [comment]
@port19x might I ask which window manager?
I consider switching every time I need to reinstall Qtile. =|

**Reply by port19x** [reply]
Awesomewm.
It has better resizing and widgets and is the most packaged tiling wm.
Qtile does multi monitors better imo and has better docs as well as a more readable config file.
You may feel it is bloated, as it comes with a notification daemon, a bar, a rightclick menu, wallpaper setter and probably a few more utilities that might make it a tiling desktop environment rather than just a wm.
Make sure to do plenty of research before switching and contemplate a few alternatives
